domain/src/todolist_scalar.rs

```diff
@@ -1,43 +1,38 @@
+use framework::*;
 use serde::{Deserialize, Serialize};
 
 #[derive(Copy, Clone, Default, Debug, PartialEq, Eq, Serialize, Deserialize, Hash)]
 pub struct TaskIndex(pub usize);
 
-impl From<usize> for TaskIndex {
-    fn from(value: usize) -> Self {
-        Self(value)
+impl From<&usize> for TaskIndex {
+    fn from(value: &usize) -> Self {
+        Self(*value)
     }
 }
 
-impl From<TaskIndex> for usize {
-    fn from(value: TaskIndex) -> Self {
-        value.0
-    }
-}
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct TaskName(String);
 
-impl std::ops::AddAssign<usize> for TaskIndex {
-    fn add_assign(&mut self, rhs: usize) {
-        self.0 += rhs
-    }
+#[derive(Debug, Error)]
+pub enum TaskNameError {
+    #[error("Task name cannot be empty")]
+    TaskNameCannotBeEmpty,
 }
 
-#[derive(Clone, Debug, Serialize, Deserialize)]
-pub struct TaskName(pub String);
-
-impl TryFrom<String> for TaskName {
-    type Error = String;
+impl TryFrom<&String> for TaskName {
+    type Error = TaskNameError;
 
-    fn try_from(value: String) -> Result<Self, Self::Error> {
+    fn try_from(value: &String) -> Result<Self, Self::Error> {
         if value.is_empty() {
-            return Err("Task name cannot be empty".to_string());
+            return Err(TaskNameError::TaskNameCannotBeEmpty);
         }
 
-        Ok(Self(value))
+        Ok(Self(value.to_owned()))
     }
 }
 
-impl From<TaskName> for String {
-    fn from(value: TaskName) -> Self {
-        value.0
+impl Into<String> for &TaskName {
+    fn into(self) -> String {
+        self.0.clone()
     }
```

Comment:
**Suggestion:** Consider using `Into<String>` trait for `TaskName` instead of `&TaskName`. This will make the code more idiomatic and easier to use. [enhancement]
```suggestion
impl Into<String> for TaskName {
    fn into(self) -> String {
        self.0
    }
}
```